subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CompilationOutputsFixture.groovy

```diff
@@ -38,12 +46,20 @@ class CompilationOutputsFixture {
         T result = operation?.call()
         snapshot.clear()
         targetDir.eachFileRecurse(FileType.FILES) {
-            it.lastModified = 0
-            snapshot << it
+            if (isIncluded(it)) {
+                it.lastModified = 0
+                snapshot << it
+            }
         }
         result
     }
 
+    private boolean isIncluded(File file) {
+        includeExtensions.empty || includeExtensions.any {
+            FileUtils.hasExtensionIgnoresCase(file.name, it)
```

Comment:
nitpick - we might also update the comment on the constructor to make it clear that it does not consider case when checking the extension.

Reply:
done